tools/autograd/templates/Functions.cpp

```diff
@@ -364,17 +364,23 @@ Tensor unsqueeze_to(const Tensor & self, int64_t dim, IntList sizes) {
   return self;
 }
 
-std::vector<Tensor> cat_tensors_backward(const Tensor & grad, const std::vector<int64_t> &sizes, int64_t dim) {
+std::vector<Tensor> cat_tensors_backward(const Tensor & grad, const std::vector<std::vector<int64_t>> &sizes, int64_t dim) {
+  if (sizes.size() > 0) {
+    // cat wraps dim to the first tensor's shape 
+    dim = at::maybe_wrap_dim(dim, sizes[0].size());
+  }
   std::vector<Tensor> grad_inputs(sizes.size());
   int64_t accumulate = 0;
   for (size_t i = 0; i < sizes.size(); ++i) {
-    auto size = sizes[i];
-    accumulate += size;
-    if (size == 0) {
+    auto& shape = sizes[i];
+    // If input was empty tensor, gradInput should be empty tensor.
+    if (shape[0] == 0) {
```

Comment:
shape == {0} seems better, will be easier to find/replace when we support full empty shapes.

Reply:
`shape == {0}` doesn't compile, should I try `shape == std::vector<int>({0})` instead?

Reply:
Ya the assumption was sizes[o] would be an IntList due to the above comment, in which case IntList({0}) should work.